### PR TITLE
Communication spec draft.

### PR DESCRIPTION
**Pull request by t510599**

### CODE REVIEW

File: src/CommunicationSpec.md [hunk 1]
```diff
@@ -1,98 +1,619 @@
...
+    "hp": integer,
+    "atk": integer,
+    "effects": [effect]
+}
+```
+
+| Field | Type | Description |
+| --- | --- | --- |
+| position | [position](#position) | Position of the object. |
+| owner_type | string | Owner of the object. <br>Value: `player` or `enemy`. |
```
Comment (secminhr) [inline src/CommunicationSpec.md]: field name inconsistency

File: src/CommunicationSpec.md [hunk 2]
```diff
@@ -1,98 +1,619 @@
...
+Send when client is ready for a game.
 
-### enter_random
-Send when client wants to join a random room.
+```jsonc
+{
+    "id": "ready_start",
+    "data": {
+		"deck_id": 1
+	}
```
Comment (secminhr) [inline src/CommunicationSpec.md]: bad indent

File: src/CommunicationSpec.md [hunk 3]
```diff
@@ -1,98 +1,619 @@
...
+    }
+}
+```
+
+| Field | Type | Description |
+| --- | --- | --- |
+| cards | integer | Serveral card IDs that the client wants to keep. |
+
+### play_card
+Prerequisite: game has started and the player is qualified for doing action.
```
Comment (secminhr) [inline src/CommunicationSpec.md]: an action/actions

File: src/CommunicationSpec.md [hunk 4]
```diff
@@ -1,98 +1,619 @@
...
+
+Send when client move an object.
+
+```jsonc
+{
+    "id": "move",
+    "data": {
+        "original_position": [x1, y1],
+		"target_position": [x2, y2]
```
Comment (secminhr) [inline src/CommunicationSpec.md]: bad indent

File: src/CommunicationSpec.md [hunk 5]
```diff
@@ -1,98 +1,619 @@
...
+Prerequisite: game has started and the player is qualified for doing action.
+
+Send when client wants to use his object to attack another object.
+
+```jsonc
+{
+    "id": "attack",
+    "data": {
+        "original_position": [x1, y1],
+		"target_position": [x2, y2]
```
Comment (secminhr) [inline src/CommunicationSpec.md]: bad indent

File: src/CommunicationSpec.md (hunk 2 above)
Comment (t510599) [reply src/CommunicationSpec.md]: SAD tab

Comment (secminhr) [reply src/CommunicationSpec.md]: SADD

**Comment by t510599** [comment]
Fixed.

**Review by spacezipper — APPROVED** [review]

**Pull request merged**